app/code/Magento/Customer/Block/Account/AuthenticationPopup.php

```diff
@@ -48,6 +59,16 @@ public function __construct(
      */
     public function getJsLayout()
     {
+        // Check if captcha is not enabled and user is not logged in
+        if (!$this->_scopeConfig->getValue(
+            Form::XML_PATH_CUSTOMER_CAPTCHA_ENABLED,
+            ScopeInterface::SCOPE_STORE
+        ) && !$this->isLoggedIn()) {
+            if (isset($this->jsLayout['components']['authenticationPopup']['children']['captcha'])) {
+                unset($this->jsLayout['components']['authenticationPopup']['children']['captcha']);
+            }
+        }
```

Comment:
The logic you've just added is related to the captha extension, as it's added in that module in https://github.com/magento/magento2/blob/df37b871ebd2a7e23ae6dec8c084855a65fbcab1/app/code/Magento/Captcha/view/frontend/layout/default.xml#L10-L27

I would suggest doing the following:
1. add layoutProcessors as optional array option to constructor
2. Adjust getJsLayout function to something like this
```suggestion
        foreach ($this->layoutProcessors as $processor) { 
            $this->jsLayout = $processor->process($this->jsLayout); 
        }
```
This will make code as close as possible to https://github.com/magento/magento2/blob/c5614048eaaff80c1d4786b391e0c3cc55c7417c/app/code/Magento/Checkout/Block/Onepage.php#L78-L85
3. In the Capcha extension, create own layout processor for AuthenticationPopup and **add** all needed parameters through this class + remove the configuration code from layout.xml

**OR**
Do some plugin for this specific method in the captcha extension, and add needed parameters based on the configuration